Feature/assignment

subinasr opened the pull request.

subinasr committed:
Assignment Section in Home Page

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+const maxItemsPerPage = 5;
+
+function assignmentRenderer(info: AssignmentRendererProps) {
+    const handleClick = () => {
```
use useCallback

AdityaKhatri replied:
Also deconstruct info access have id and handleClick

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+                href={emptyLink}
+                className={styles.link}
+            >
+                {info.projectDetails.title}
```
check if projectDetails is defined

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+function Assignments(props: Props) {
+    const [activePage, setActivePage] = useState<number>(1);
+    const [assignmentCount, setAssignmentCount] = useState<number>(0);
+    const [selectedAssignment, setSelectedAssignment] = useState<number>();
```
The type should be number | undefined

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+            url: 'server://assignments/',
+            method: 'GET',
+            query: {
+                is_done: 3,
```
What does `3` mean?

subinasr replied:
Unfortunately, instead of 'true' or 'false', is_done is assigned numbers (1: unknown, 2: true, 3: false) from the server itself.

AdityaKhatri replied:
Could you add a comment in the code on what the numbers mean

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+        },
+    );
+
+    const rendererParams = (id: number, info: Assignment) => (
```
use UseCallback

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+        ,
+        ,
+        triggerBulkAsDone,
+    ] = useRequest<unknown>(
```
Define type for response

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+const emptyLink = '#';
+const maxItemsPerPage = 5;
+
+function assignmentRenderer(info: AssignmentRendererProps) {
```
Component should be capitalized: AssignmentRenderer
```suggestion
function AssignmentRenderer(info: AssignmentRendererProps) {
```

subinasr committed:
Mark As Done

- Bulk Mark As Done
- Mark Assignment Instances as Done

subinasr committed:
Comment Fixes:

- use Callback for rendererParams and assignmentRenderer
- type fixes
- remove unnecessary imports
- handle disabled for buttons

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+
+import ListView from '#rsu/../v2/View/ListView';
+import FormattedDate from '#rscv/FormattedDate';
+import Icon from '#rscg/Icon';
```
Icon is not used. We should remove unused imports

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+    };
+}
+
+interface Props {
+
+}
```
remove this

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+const keySelector = (info: Assignment) => info.id;
+
+
+function Assignments(props: Props) {
```
```suggestion
function Assignments() {
```

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+
+function Assignments(props: Props) {
+    const [activePage, setActivePage] = useState<number>(1);
+    const [assignmentCount, setAssignmentCount] = useState<number>(0);
```
this state value is not used. we should remove this.

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+    const [assignmentCount, setAssignmentCount] = useState<number>(0);
+    const [selectedAssignment, setSelectedAssignment] = useState<number | undefined>();
+    const [
+        pending,
```
maybe we should show loading animation when pending is true

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+    );
+
+    const [
+        markAsDonePending,
```
we can use this pending value to disable the button that triggers this request

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+        },
+    );
+
+    const rendererParams = useCallback(
+        (_: number, info: Assignment) => ({
+            ...info,
+            handleClick: setSelectedAssignment,
+        }), [setSelectedAssignment]);
```
```suggestion
    const rendererParams = useCallback((_: number, info: Assignment) => ({
        ...info,
        handleClick: setSelectedAssignment,
    }), [setSelectedAssignment]);
```

samshara commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,223 @@
+                <h2 className={styles.heading}>
+                    {_ts('assignment', 'myAssignments')}
+                </h2>
+                {assignmentsResponse && assignmentsResponse.count > 0 && (
```
```suggestion
                {assignmentsResponse && assignmentsResponse?.count > 0 && (
```

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,224 @@
+        ...info,
+        handleClick: setSelectedAssignment,
+        markAsDonePending,
+    }), [setSelectedAssignment]);
```
```suggestion
    }), [setSelectedAssignment, markAsDonePending]);
```

AdityaKhatri commented on src/views/Home/Assignment/index.tsx:
```diff
@@ -0,0 +1,224 @@
+        {
+            url: `server://assignments/${selectedAssignment}/`,
+            method: 'PUT',
+            body: { is_done: true },
```
Aren't we going with 1, 2 and 3 here?

subinasr replied:
No. In body is_done has true/false value while in url it takes on integer value of either 1, 2 or 3.

AdityaKhatri committed:
Fix assignment styling

samshara reviewed (changes requested):
please fix deepscan issues
